### DISCUSSION TITLE
Would it be possible to deploy Mesop as a serverless function

### DISCUSSION THREAD

**Discussion by jmfk** [google/mesop#1120] [Q&A] [answered]
Would it be possible to deploy Mesop as a serverless function. 
Given the stateless nature this should in theory be possible, right?
But how hard would it be to to do?

**Comment by richard-to** [comment, marked as answer]
Yes, I think it could be possible to do.

If I understand your question correctly, you want to host say the HTML/JS on a separate server. Then use Mesop as the rendering server and host that as a Cloud Functions (or other non-Google alternative) API endpoint.

I think for GCP Cloud Functions or AWS Lambda, you'd need to see if SSE's are supported since Mesop uses SSEs when sending the ui requests.

I asked ChatGPT, and neither seem to support to SSEs apparently, but you should verify for sure.

Another option is to use something more heavyweight like Cloud Run.

-------

I think to make it work, the JS bundle would just need to be edited so that it points the remote server. Probably need to handle CORs in this case.

**Reply by jmfk** [reply]
Cool. I'll check out SSE and if that works with AWS lambdas (which is what I be using in that case).

**Reply by jmfk** [reply]
Seems like AWS Lambda might not work that great with SSE (since it returns a function result). There might be other AWS solution that could work.
